src/connector/src/sink/remote.rs

```diff
@@ -228,26 +241,145 @@ async fn validate_remote_sink(param: &SinkParam) -> Result<()> {
     .map_err(|e| anyhow!("unable to validate: {:?}", e))?
 }
 
+enum StreamChunkConverter {
+    Es(EsStreamChunkConverter),
+    Other,
+}
+impl StreamChunkConverter {
+    fn new(
+        sink_name: &str,
+        schema: Schema,
+        pk_indices: &Vec<usize>,
+        properties: &HashMap<String, String>,
+    ) -> Result<Self> {
+        if sink_name.eq(ElasticSearchSink::SINK_NAME) {
+            Ok(StreamChunkConverter::Es(EsStreamChunkConverter::new(
+                schema,
+                pk_indices.clone(),
+                properties.get(ES_OPTION_DELIMITER).cloned(),
+            )?))
+        } else {
+            Ok(StreamChunkConverter::Other)
+        }
+    }
+
+    fn convert_chunk(&self, chunk: StreamChunk) -> Result<StreamChunk> {
+        match self {
+            StreamChunkConverter::Es(es) => es.convert_chunk(chunk),
+            StreamChunkConverter::Other => Ok(chunk),
+        }
+    }
+}
+struct EsStreamChunkConverter {
```

Comment:
This file is already very large. Can we move the es specific logic, such as the converter, to a separate file (e.g. elastic_search.rs)?